src/taipy/core/data/data_node.py

```diff
@@ -181,10 +181,13 @@ def validity_period(self, val):
     @property  # type: ignore
     @_self_reload(_MANAGER_NAME)
     def expiration_date(self) -> datetime:
-        if not self._last_edit_date:
+        last_edit_date = self.last_edit_date
+        validity_period = self._validity_period
+
+        if not last_edit_date:
             raise NoData(f"Data node {self.id} has not been written yet.")
 
-        return self._last_edit_date + self.validity_period if self.validity_period else self._last_edit_date
```

Comment:
Here, we called the getter for `validity_period` but not `last_edit_date`.

Reply:
I don't have an answer for this as I think _validity_date should have been enough :D